cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/spark/SparkProgramRunnerTest.java

```diff
@@ -306,6 +297,61 @@ public void apply() throws Exception {
   }
 
   @Test
+  public void testSparkWithTimePartitionedFileSetAndDSArgs() throws Exception {
+    testSparkWithTimePartitionedFileSetAndDSArgs(SparkAppUsingFileSet.class, SparkAppUsingFileSet.JavaCharCount.class);
```

Comment:
Instead of creating two new tests, can you modify the app instead so that it do two rounds of read/write, such that one round with dataset arguments and one without?

Adding new test methods for this case is not necessary and it gives a higher cost (hence longer run time) for running unit-tests.